proton-j/src/main/java/org/apache/qpid/proton/engine/impl/CollectorImpl.java

```diff
@@ -57,9 +58,13 @@ public void pop()
         }
     }
 
-    public EventImpl put(Event.Type type, Object context)
+    public EventImpl put(EventType type, Object context)
     {
-        if (tail != null && tail.getType() == type &&
+        if (type == null)
+            throw new IllegalArgumentException("Type cannot be null");
+        if (!type.isValid())
+            throw new IllegalArgumentException("Cannot put events of type " + type);
+        if (tail != null && tail.getEventType() == type &&
```

Comment:
Would an equals() check be more appropriate now that EventType isn't necessarily an Enum (though granted it is natural and we know it is for the core events)?